src/lib/dataset/pandas.ts

```diff
@@ -1,115 +1,71 @@
-      return 'integer';
-    } else if (_isFloat(val)) {
-      return 'float';
-    }
-  } else if (type === 'object' && val instanceof Date) {
-    return 'date';
-  } else if (_isSupportedType(type) && val !== null) {
-    return type;
-  }
-
-  return null;
+const COL_TYPE_MAPPING: Record<TableSchemaType, DataSetColumnType> = {
+  integer: 'integer',
+  number: 'float',
+  boolean: 'boolean',
+  datetime: 'date',
+  duration: 'date',
+  any: 'object',
+  string: 'string',
+};
+
+function tableSchemaTypeToDataSetColumnType(tableSchemaType: TableSchemaType): DataSetColumnType {
+  return COL_TYPE_MAPPING[tableSchemaType];
 }
 
 /**
- * @descritpion extend a `DataSet` headers with inferred types, base on value of columns
- * @param {DataSet} dataset - The dataset that we want to extend headers with inferred types
- * @param {number} [maxRows=50] - Determine on how many rows we're goind to do our guess
+ * Transform a pandas dataframe, exported with `.to_json(orient='table')` into * a `DataSet` structure
  *
+ * The structure of the `orient='table'` format is:
+ * ```
+ * {
+ *   schema: { fields: [{name, type}, ...] },
+ *   data: [{column -> value}, ...]
+ * }
+ * ```
+ *
+ * See https://pandas.pydata.org/docs/user_guide/io.html#table-schema for more details on this format.
+ *
+ * TODO Handle pagination context
  */
-export function inferTypeFromDataset(dataset: DataSet, maxRows = 50): DataSet {
-  // For the moment, we infer arbitrarily column's types on maximum the 50 first rows
-  maxRows = Math.min(dataset.data.length, maxRows);
-  const newHeaders: DataSetColumn[] = [];
-
-  for (const [colidx, header] of dataset.headers.entries()) {
-    let prevType = null;
-
-    for (let j = 0; j < maxRows; j++) {
-      const currentValue = dataset.data[j][colidx];
-      const guessedType = _guessType(currentValue, prevType);
-
-      if (prevType === null) {
-        prevType = guessedType;
-      } else if (prevType !== guessedType) {
-        // if integers and floats are mixed, guessed type should be 'float'
-        const typeSet = new Set([prevType, guessedType]);
-        if (typeSet.has('integer') && typeSet.has('float')) {
-          prevType = 'float';
-        } else {
-          prevType = null;
-          break;
-        }
-      }
-    }
-
-    // We pass here if we find 2 different types in the same column
-    if (prevType === null) {
-      newHeaders.push(header);
-    } else {
-      newHeaders.push({ ...header, type: prevType as DataSetColumnType });
-    }
+export function pandasDataTableToDataset(pandasTable: PandasDataTable): DataSet {
+  const dataset: DataSet = { headers: [], data: [] };
+  dataset.headers = pandasTable.schema.fields.map(({ name, type }) => ({
+    name,
+    type: tableSchemaTypeToDataSetColumnType(type),
+  }));
+  for (const row of pandasTable.data) {
+    dataset.data.push(
+      dataset.headers.map(column =>
+        Object.prototype.hasOwnProperty.call(row, column.name) ? row[column.name] : null,
```

Comment:
C'est possible que pandas ne renvoie pas toutes les colonnes pour chaque ligne de donnée ?

Reply:
It's true that mongo definitely can, but pandas shouldn't.
I can remove this condition as it's not necessary

Reply:
Ok I just wanted to be sure I didn't miss something :+1: We can keep it